### ISSUE TITLE
custom xCommand Preset button

### ISSUE THREAD

**Issue by nottooloud** [bitfocus/companion-module-cisco-webex-websocket#5]
xConfig button works great and will be very handy.
I assume the xCommand button just isn't wired up yet?
I'd guess layout of that is a bit of a puzzle, since you can pack a lot of parameters into one command.

**Comment by AlexMPopov** [comment]
Hello.
Here is a 2 examples that worked for me:

<img width="637" alt="Screenshot 2021-05-21 at 09 12 51" src="https://user-images.githubusercontent.com/74984001/119097228-c6a8de00-ba14-11eb-8b05-0d11aee621e2.png">
<img width="647" alt="Screenshot 2021-05-21 at 09 12 24" src="https://user-images.githubusercontent.com/74984001/119097233-c7da0b00-ba14-11eb-95fd-52bf73eb3dcf.png">



`xCommand/Video/Selfview/Set
 {"FullscreenMode":"On","OnMonitorRole":"Second","Mode":"On"}

xCommand/Video/Matrix/Assign
{"Output":3,"SourceID":2,"Mode":"Add"}
`

**Comment by jruys** [comment]
Adding another example. I was tinkering with the local layout - this works for me:

![image](https://user-images.githubusercontent.com/16433735/119499313-e880c880-bd66-11eb-9953-7e33eda0c12e.png)

**Comment by nottooloud** [comment]
Spaces work in the Method, same as is noted in the xConfig Preset.
![Screen Shot 2021-05-25 at 1 11 52 PM](https://user-images.githubusercontent.com/61946064/119539928-ca14d000-bd5a-11eb-9ad3-0e4cf6a639aa.png)

**Comment by nottooloud** [comment]
The all cameras Preset Store xCommand uses Method Preset Store and Params {"PresetID":1,"Type":"Camera","Description":"your preset name"}

**Comment by bwentzel** [comment]
> xCommand

@AlexMPopov can you confirm it worked on your side with method set to “xCommand …”
Seems others only get it to work when leaving xCommand out. 
Example working: 
![908ACE1C-D433-4C5C-9C05-6E6B7338B1D2](https://user-images.githubusercontent.com/23614532/120720374-42526280-c4cc-11eb-9eff-c0b353c92dff.jpeg)

not working:  @@
![1464F260-5971-4E3D-AB43-2006FFC01B87](https://user-images.githubusercontent.com/23614532/120720419-5302d880-c4cc-11eb-969b-efe4abfbb798.jpeg)

**Comment by AlexMPopov** [comment]
@bwentzel you are correct in that it worked with xcommand. The screenshot I posted is from a working setup.

**Comment by rgischig** [comment]
Hi nottooloud,

I've worked at Cisco when TC 7.3 was new.. this was many many moon's ago. If you can get your hands on a new device wich would run a CE software, like a SX20 or later, I would argue that it should work. 
The TC series software is really old, so I don't think there is even support for webhooks. You might can use a TCL script from a Raspi to telnet into the C20, and execute the comand like that (similar to the example I've done with some PoE switches, which don't offer an API.. https://gischig.com/controling-my-cisco-sg300-switch-with-the-streamdeck-paspi/ )

roman

**Comment by SveinTerje** [comment]
This issue should be closed. The feature works perfectly on CE and RoomOS software. TC Software should be out of scope as it is end of support.
